wapitiCore/attack/mod_drupal_enum.py

```diff
@@ -30,13 +31,43 @@ def get_hashes(self):
             data = json.load(hashes)
             return data
 
+    @staticmethod
+    def calculate_git_hash(path):
+        try:
+            with open(path, 'rb') as file:
+                data = file.read()
+
+            # Calculate the size of the file
+            filesize = len(data)
+
+            # Create the string to hash to match the git hash function
+            to_hash = b"blob " + str(filesize).encode() + b"\0" + data
+
+            git_hash = hashlib.sha1(to_hash).hexdigest()
+
+            return git_hash
+        except FileNotFoundError:
+            print(f"File not found: {path}")
+            return None
+        except IOError as e:
+            print(f"IO error: {str(e)}")
+            return None
+
     async def get_url_hash(self, root_url: str, path: str) -> Tuple[str, str]:
         request = Request(f"{root_url}{path}", "GET")
         response: Response = await self.crawler.async_send(request, follow_redirects=True)
         if response.is_error:
             return "", ""
 
-        return hashlib.sha256(response.content.encode()).hexdigest(), path
+        with tempfile.TemporaryDirectory() as temp_dir:
+            filename = ntpath.basename(path)
+            file_path = path_join(temp_dir, filename)
+            with open(file_path, "wb") as file:
+                file.write(response.bytes)
```

Comment:
You should do the calculus using `response.bytes` directly. Writing each file on the disk will take too much time